match2/commons/player_display.lua

```diff
@@ -0,0 +1,96 @@
+local Class = require('Module:Class')
+local DisplayUtil = require('Module:DisplayUtil')
+local MatchGroupUtil = require('Module:MatchGroup/Util')
+local TypeUtil = require('Module:TypeUtil')
+
+local html = mw.html
+
+--[[
+Display components for players.
+]]
+local PlayerDisplay = {propTypes = {}}
+
+PlayerDisplay.propTypes.BlockPlayer = {
+	flip = 'boolean?',
+	overflow = TypeUtil.optional(DisplayUtil.types.OverflowModes),
+	player = MatchGroupUtil.types.Player,
+	showFlag = 'boolean?',
+	showLink = 'boolean?',
+}
+
+--[[
+Displays a player as a block element. The width of the component is
+determined by its layout context, and not by the player name.
+]]
+function PlayerDisplay.BlockPlayer(props)
+	DisplayUtil.assertPropTypes(props, PlayerDisplay.propTypes.BlockPlayer)
+	local player = props.player
+
+	local nameNode = html.create('span'):addClass('name')
+		:wikitext(props.showLink ~= false and player.pageName
+			and '[[' .. player.pageName .. '|' .. player.displayName .. ']]'
+			or player.displayName
+		)
+	DisplayUtil.applyOverflowStyles(nameNode, props.overflow or 'ellipsis')
+
+	local flagNode
+	if props.showFlag ~= false and player.flag then
+		flagNode = PlayerDisplay.Flag(player.flag)
+	end
+
+	return html.create('div'):addClass('block-player')
+		:addClass(props.flip and 'flipped' or nil)
+		:node(flagNode)
+		:node(nameNode)
+end
+
+PlayerDisplay.propTypes.InlinePlayer = {

```

Comment:
Also considered naming these `PlayerInline/PlayerBlock` instead of `InlinePlayer/BlockPlayer`. Let me know if you prefer the component type at the end instead.